worker/caasapplicationprovisioner/application.go

```diff
@@ -103,82 +108,134 @@ func (a *appWorker) loop() error {
 		return errors.Errorf("charm version 2 or greater required")
 	}
 
+	// Update the password once per worker start to avoid it changing to frequently.
+	a.password, err = utils.RandomPassword()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	err = a.facade.SetPassword(a.name, a.password)
```

Comment:
This should not be done every worker restart - it should be done once per deploy. ie when the application watcher first notices the application

Reply:
I don't think this is a problem. The worker needs to know the password and I prefer it have ownership over the password lifecycle, since it is the thing the ensures the provider updates the Secret.

This password is only the application password used by the init container, so it isn't an issue if it changes briefly to be replaced in the Secret.